ccl/schemachangerccl: TestPauseMixedVersion_ccl_alter_table_alter_primary_key_rbr failed

ccl/schemachangerccl.TestPauseMixedVersion_ccl_alter_table_alter_primary_key_rbr [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21000392?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21000392?buildTab=artifacts#/) on release-25.3 @ [09f2b47406eb506426d000ebf590245d03d2d030](https://github.com/cockroachdb/cockroach/commits/09f2b47406eb506426d000ebf590245d03d2d030):


```
test logs left over in: /artifacts/tmp/_tmp/47b2f256c3c66dd3a9f82a3dafa4441d/logTestPauseMixedVersion_ccl_alter_table_alter_primary_key_rbr3295234100
--- FAIL: TestPauseMixedVersion_ccl_alter_table_alter_primary_key_rbr (121.56s)
=== RUN   TestPauseMixedVersion_ccl_alter_table_alter_primary_key_rbr/post_commit_stage_1_of_24
    cumulative.go:403: job 1143176388709253121 is paused
    cumulative.go:410: job 1143176388709253121 is resuming
    sql_runner.go:117: 
        	Error Trace:	pkg/testutils/sqlutils/sql_runner.go:130
        	            				pkg/testutils/sqlutils/sql_runner.go:391
        	            				pkg/sql/schemachanger/sctest/cumulative.go:412
        	            				pkg/ccl/schemachangerccl/multiregion_testcluster_factory.go:87
        	            				pkg/sql/schemachanger/sctest/cumulative.go:417
        	            				pkg/sql/schemachanger/sctest/cumulative.go:351
        	            				pkg/sql/schemachanger/sctest/framework.go:721
        	Error:      	Received unexpected error:
        	            	query 'SELECT status, error FROM [SHOW JOB 1143176388709253121]': expected:
        	            	(1) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).CheckQueryResultsRetry.func1
        	            	  | 	pkg/testutils/sqlutils/sql_runner.go:394
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithinError.func1
        	            	  | 	pkg/testutils/soon.go:71
        	            	  | github.com/cockroachdb/cockroach/pkg/util/retry.ForDuration
        	            	  | 	pkg/util/retry/retry.go:236
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithinError
        	            	  | 	pkg/testutils/soon.go:77
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).succeedsWithin
        	            	  | 	pkg/testutils/sqlutils/sql_runner.go:130
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).CheckQueryResultsRetry
        	            	  | 	pkg/testutils/sqlutils/sql_runner.go:391
        	            	  | github.com/cockroachdb/cockroach/pkg/sql/schemachanger/sctest.pause.func2
        	            	  | 	pkg/sql/schemachanger/sctest/cumulative.go:412
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/schemachangerccl.MultiRegionTestClusterFactory.Run
        	            	  | 	pkg/ccl/schemachangerccl/multiregion_testcluster_factory.go:87
        	            	  | github.com/cockroachdb/cockroach/pkg/sql/schemachanger/sctest.pause
        	            	  | 	pkg/sql/schemachanger/sctest/cumulative.go:417
        	            	  | github.com/cockroachdb/cockroach/pkg/sql/schemachanger/sctest.PauseMixedVersion.func1
        	            	  | 	pkg/sql/schemachanger/sctest/cumulative.go:351
        	            	  | github.com/cockroachdb/cockroach/pkg/sql/schemachanger/sctest.cumulativeTestForEachPostCommitStage.func1.2
        	            	  | 	pkg/sql/schemachanger/sctest/framework.go:721
        	            	  | testing.tRunner
        	            	  | 	GOROOT/src/testing/testing.go:1690
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_arm64.s:1223
        	            	Wraps: (2) query 'SELECT status, error FROM [SHOW JOB 1143176388709253121]': expected:
        	            	  | succeeded, 
        	            	  |
        	            	  | got:
        	            	  | running, 
        	            	Error types: (1) *withstack.withStack (2) *errutil.leafError
--- FAIL: TestPauseMixedVersion_ccl_alter_table_alter_primary_key_rbr/post_commit_stage_1_of_24 (68.62s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestPauseMixedVersion_ccl_alter_table_alter_primary_key_rbr.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-58883

--- a/pkg/sql/schemachanger/sctest/cumulative.go
+++ b/pkg/sql/schemachanger/sctest/cumulative.go
@@ -414,8 +414,12 @@ func pause(t *testing.T, factory TestServerFactory, cs CumulativeTestCaseSpec) {
 			// Resume the job and check that it succeeds.
 			tdb.Exec(t, "RESUME JOB $1", jobID)
 			t.Logf("job %d is resuming", jobID)
+			// Wait for the schema change to complete.
+			// Note: This has a longer timeout than normal succeeds soon.
+			waitForSchemaChangesToFinish(t, tdb)
+			// Validate the terminal status of the schema change.
 			qStatusWithError := fmt.Sprintf(`SELECT status, error FROM [SHOW JOB %d]`, jobID)
-			tdb.CheckQueryResultsRetry(t, qStatusWithError, [][]string{{"succeeded", ""}})
+			tdb.CheckQueryResults(t, qStatusWithError, [][]string{{"succeeded", ""}})
 		}
 		waitForSchemaChangesToFinish(t, tdb)
 		require.Equal(t, uint32(1), atomic.LoadUint32(&numInjectedFailures))

